src/main/java/com/sforce/ws/bind/XmlObject.java

```diff
@@ -302,6 +330,42 @@ protected void loadAfterStartTag(XmlInputStream in, TypeMapper typeMapper) throw
         if (textFound) value = typeMapper.deserialize(text.toString(), xmlType);
     }
 
+    private XmlObject extractChildElement(XmlInputStream in, TypeMapper typeMapper) throws ConnectionException {
```

Comment:
TypeMapper does have something similar in readSingle's logic but it has more information passed in than what we have access to here.